enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/utils/default-embed-setting.ts

```diff
@@ -1,26 +1,45 @@
 import { match } from "ts-pattern";
 
+import type {
+  DashboardEmbedOptions,
+  ExplorationEmbedOptions,
+  QuestionEmbedOptions,
+} from "metabase-enterprise/embedding_iframe_sdk/types/embed";
+
 import type { SdkIframeEmbedSetupExperience } from "../types";
 
 export const getDefaultSdkIframeEmbedSettings = (
   type: SdkIframeEmbedSetupExperience,
-  defaultEntityId: string | number,
+  defaultResourceId: string | number,
 ) =>
   match(type)
-    .with("dashboard", () => ({
-      dashboardId: defaultEntityId,
-      isDrillThroughEnabled: true,
-      withDownloads: false,
-      withTitle: true,
-    }))
-    .with("chart", () => ({
-      questionId: defaultEntityId,
-      isDrillThroughEnabled: true,
-      withDownloads: false,
-      withTitle: true,
-    }))
-    .with("exploration", () => ({
-      template: "exploration",
-      isSaveEnabled: true,
-    }))
+    .with(
+      "dashboard",
+      (): DashboardEmbedOptions => ({
+        dashboardId: defaultResourceId,
+        isDrillThroughEnabled: true,
+        withDownloads: false,
+        withTitle: true,
+      }),
+    )
+    .with(
+      "chart",
+      (): QuestionEmbedOptions => ({
+        questionId: defaultResourceId,
+        isDrillThroughEnabled: true,
+        withDownloads: false,
+        withTitle: true,
+      }),
+    )
+    .with(
+      "exploration",
+      (): ExplorationEmbedOptions => ({
+        template: "exploration",
+        isSaveEnabled: true,
+      }),
+    )
     .exhaustive();
+
+// TODO: will be replaced with the most recently visited dashboard once EMB-508 is merged.
+export const DEFAULT_SDK_IFRAME_EMBED_SETTINGS =
```

Comment:
When you made this dynamic, I don't think we should keep this name. A constant is a value that's never changed or at least rarely changes. But when we use the most recently visited dashboard, this value could change at any given moment. It shouldn't be named like a constant anymore.